src/org/jgroups/raft/filelog/FilePositionCache.java

```diff
@@ -11,77 +12,222 @@
 public class FilePositionCache {
 
    public static final int TOO_OLD = -1;
-   public static final int NO_CAPACITY = -2;
-   public static final int OK = 1;
-   private static final int SIZE_INCREMENT = 128;
 
-   private final long[] position;
+   private static final int PAGE_CAPACITY = 1 << 10;
```

Comment:
suggestion: make this configurable.
`RAFT` allows some arguments to be set in the `Log` using the configuration. 

[1] https://github.com/belaban/jgroups-raft/blob/717d502b5b0dedcb485500097975400861c41107/src/org/jgroups/protocols/raft/RAFT.java#L450

Reply:
Let me add it as a configurable parameter then, although it's a parameter that won't make easier the life of users and would be rather better to autotune it...
The default existing snapshotting policy for log files it's based on log file current length and not on the remaining "caching" memory (that's a way more precious resource then disk, depending on the use cases): if it would be different, then the cache itself could be simplified back to its original form, because can be pre-allocated at the max expected capacity and recycled over and over (@pruivo )

Reply:
if we can do autotune, just leave the code as-is and we can do it later :+1: